### PR TITLE
Added flake8 to pre-commits

### PR DESCRIPTION
**Pull request by sammaule**
Adding flake8 to the pre-commits. Suggest using numpy as the docstring convention.

### CODE REVIEW

**Comment by longm89** [comment]
It's working well for me.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[flake8]
+select = ANN,B,B9,BLK,C,D,E,F,I,S,W
+  # only check selected error codes
+application-import-names = horizon_net
```
Comment (anaprietonem) [inline .flake8:4]: I have a quick Q here, my understanding is that this field refers to the folders that flake8 will check. But I can't see any folder called `horizon_net`, so I was wondering how are we testing that this is working?

Comment (sammaule) [reply .flake8:4]: Got ahead of things slightly here. Will be having a horizon_net folder in the repo containing the model code that will be imported by the Lambda func API. This just tells flake8 that this is a local import https://pypi.org/project/flake8-import-order/

Comment (anaprietonem) [reply .flake8:4]: Okey, then I guess happy to approve

**Review by anaprietonem — APPROVED** [review]

**Pull request merged**